neo/utils.py

```diff
@@ -0,0 +1,856 @@
+    """
+    Internal function
+    """
+    # wrap annotation value to be list
+    if not type(annotation_value) in [list, np.ndarray]:
+        annotation_value = [annotation_value]
+
+    # get all real attributes of object
+    attributes = inspect.getmembers(obj)
+    attributes_names = [t[0] for t in attributes if not(
+        t[0].startswith('__') and t[0].endswith('__'))]
+    attributes_ids = [i for i, t in enumerate(attributes) if not(
+        t[0].startswith('__') and t[0].endswith('__'))]
+
+    # check if annotation is present
+    value_avail = False
+    if annotation_key in obj.annotations:
+        check_value = obj.annotations[annotation_key]
+        value_avail = True
+    elif annotation_key in obj.array_annotations:
+        check_value = obj.array_annotations[annotation_key]
+        value_avail = True
+    elif annotation_key in attributes_names:
+        check_value = attributes[attributes_ids[
+            attributes_names.index(annotation_key)]][1]
+        value_avail = True
+
+    if value_avail:
+        # check if annotation is list and fits to length of object list
+        if not _is_annotation_list(check_value, len(obj)):
+            # check if annotation is single value and fits to requested value
+            if check_value in annotation_value:
+                valid_mask = np.ones(obj.shape)
+            else:
+                valid_mask = np.zeros(obj.shape)
+                if type(check_value) != str:
+                    warnings.warn(
+                        'Length of annotation "%s" (%s) does not fit '
+                        'to length of object list (%s)' % (
+                            annotation_key, len(check_value), len(obj)))
+
+        # extract object entries, which match requested annotation
+        else:
+            valid_mask = np.zeros(obj.shape)
+            for obj_id in range(len(obj)):
+                if check_value[obj_id] in annotation_value:
+                    valid_mask[obj_id] = True
+    else:
+        valid_mask = np.zeros(obj.shape)
+
+    valid_ids = np.where(valid_mask)[0]
+
+    return valid_ids
+
+
+def _get_valid_annotations(obj, valid_ids):
+    """
+    Internal function
+    """
+    sparse_annotations = copy.deepcopy(obj.annotations)
+    for key in sparse_annotations:
+        if _is_annotation_list(sparse_annotations[key], len(obj)):
+            sparse_annotations[key] = list(np.array(sparse_annotations[key])[
+                valid_ids])
+    return sparse_annotations
+
+
+def _is_annotation_list(value, exp_length):
+    """
+    Internal function
+    """
+    return (
+        (isinstance(value, list) or (
+            isinstance(value, np.ndarray) and value.ndim > 0)) and (len(value) == exp_length))
+
+
+def add_epoch(
+        segment, event1, event2=None, pre=0 * pq.s, post=0 * pq.s,
+        attach_result=True, **kwargs):
+    """
+    Create epochs around a single event, or between pairs of events. Starting
```

Comment:
"Epoch" and "Event" should be capitalized throughout the docstring.

Reply:
I went through the whole file and made sure that I consistently call the actual objects Epochs and Events while referring to single events/epochs they contain by the lower-case names.